src/VisualStudio/IntegrationTest/TestUtilities/VisualStudioInstanceFactory.cs

```diff
@@ -91,13 +70,16 @@ private static Assembly AssemblyResolveHandler(object sender, ResolveEventArgs e
         /// <summary>
         /// Returns a <see cref="VisualStudioInstanceContext"/>, starting a new instance of Visual Studio if necessary.
         /// </summary>
-        public VisualStudioInstanceContext GetNewOrUsedInstance(ImmutableHashSet<string> requiredPackageIds)
+        public VisualStudioInstanceContext GetNewOrUsedInstance(ImmutableHashSet<string> requiredPackageIds, Action<string> logAction = null)
         {
             ThrowExceptionIfAlreadyHasActiveContext();
 
-            bool shouldStartNewInstance = ShouldStartNewInstance(requiredPackageIds);
-            UpdateCurrentlyRunningInstance(requiredPackageIds, shouldStartNewInstance);
+            _firstChanceExceptionLogger?.Dispose();
+            _firstChanceExceptionLogger = new FirstChanceExceptionLogger(CaptureTestNameAttribute.CurrentName, logAction);
```

Comment:
*That's* how you get the name of the current test in xUnit? I spent so much time looking for that...